sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_compiler.asciidoc

```diff
@@ -461,51 +597,301 @@ is `name`.
 
 _Throws:_
 
+* An `exception` with the `errc::invalid` error code if
+  `ext_oneapi_has_kernel(name)` returns `false`.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+std::string ext_oneapi_get_raw_kernel_name(const std::string &name)
+----
+!====
+
+_Constraints:_ This function is not available when `State` is
+`bundle_state::ext_oneapi_source`.
+
+_Returns:_ If the kernel bundle was created from a bundle of state
+`bundle_state::ext_oneapi_source` and `name` was registered via
+`registered_kernel_names`, returns the compiler-generated (e.g. mangled) name
+for this kernel function.
+If the kernel bundle was created from a bundle of state
+`bundle_state::ext_oneapi_source` and `name` is the same as a
+compiler-generated name for a kernel defined in that bundle, that same
+`name` is returned.
+
+_Throws:_
+
 * An `exception` with the `errc::invalid` error code if
   `ext_oneapi_has_kernel(name)` returns `false`.
 |====
 
+=== Obtaining a kernel when the language is `sycl`
 
-== Example
+When the kernel is defined in the language `source_language::sycl`, the host
+code may query for the kernel or obtain the `kernel` object using either the
+kernel's name as it is generated by the compiler (i.e. the C++ mangled name) or
+by using the `registered_kernel_names` property.
+
+==== Using the compiler-generated name
+
+If the kernel is declared as `extern "C"`, the compiler generates the kernel
+name exactly as it appears in the source code (i.e. there is no name mangling).
+Therefore, it is easy to query for the kernel by using the compiler-generated
+name.
+For example, if the kernel is defined like this in the source code string:
+
+```
+std::string source = R"""(
+  #include <sycl/sycl.hpp>
+  namespace syclex = sycl::ext::oneapi::experimental;
+
+  extern "C"
+  SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclex::range_kernel<1>))
```

Comment:
In this design, `foo` doesn't accept `sycl::id<>` or `sycl::item<>` as a parameter, but instead is expected to use the free functions ( for example: `sycl::ext::oneapi::experimental::this_id<1>();` )  to get the "index".

So, AFAICT at my initial stage of exploring this, if that's true, then there is no need for any argument for `SYCL_EXT_ONEAPI_FUNCTION_PROPERTY`.   If it were just `SYCL_EXT_ONEAPI_KERNEL_FUNCTION` that might be enough to designate the following function as something that needs to be wrapped in a kernel call.

Reply:
We actually had a long debate about this.  We'd prefer to use the common syntax of compile-time properties, rather than inventing new macros.  The macro `SYCL_EXT_ONEAPI_FUNCTION_PROPERTY` is part of the common syntax for compile-time properties.

FWIW, see 1b6f57ec0af283567c57b7bb9584af7898214b23 for the way it looked before we decided to use `SYCL_EXT_ONEAPI_FUNCTION_PROPERTY`.